### DISCUSSION TITLE
Will we always need to use a terminal to upload data to DANDI?

### DISCUSSION THREAD

**Discussion by lmorgan6** [dandi/helpdesk#36] [Q&A]
We have some members of our team who generate data but are unfamiliar with working in a terminal or using Python. How would you recommend solving their ability to upload their own data to DANDI? Right now, they have to share the data with our site and then I upload the data for them. A GUI with a drag and drop feature for their files would be a better option for them I think. Are there any plans for something like this?

**Comment by satra** [comment]
We will be working on a graphical interface application that will mimic the CLI operation. However, this is scheduled for next year. In the short term, we can help with people who are having difficulties.

**Comment by bendichter** [comment]
I just wanted to follow up on this. @lmorgan6 , I'd be happy to schedule a time to help people on your team with this. Feel free to schedule a time [here](https://calendly.com/ben-dichter/) if you are interested.

**Comment by lmorgan6** [comment]
Hi Ben,

Thank you for following up on this -  I'll CC the team I was thinking of when I started this thread. Patrick - if you'd like to take the DANDI helpdesk up on their offer, I think they could help to set up a system where your team can self-upload the data produced at your site. Last time we had results to upload, Jessie sent them to me and I uploaded them to DANDI. I think as you produce more and more results, it could be more efficient to have your team know how to self-publish your data.

If you think this would be helpful, I think anyone who wants to learn can use the link below to schedule some time with DANDI to troubleshoot a system that will work well for your team in the future. Feel free to include Jocelyn and I and we'll help as much as we can with what we know too.

Leah
________________________________

**Comment by lmorgan6** [comment]
Hi,

I agree it makes the most sense for us to upload our own data!  I attended the DANDI workshop on the 1st, but got lost once the terminal use began.  Is using a terminal the only way to upload data?  Perhaps if I can read discussion #36 that may help--how might I be able to access that? I've scheduled an appointment with Ben for tomorrow at 1:30pm, but reading that may help prevent Ben from repeating himself.  :)

Kind regards,
Bridget

Bridget Wicinski (she/her/hers)
Lab supervisor for Dr. Patrick Hof
Nash Family Dept. of Neuroscience
Icahn School of Medicine at Mount Sinai
1425 Madison Ave, Icahn 9-26
New York, NY 10029
Phone: 212-659-8188
*********************************************************************************
"Smooth seas do not make skillful sailors."
     - African proverb  ~~_/) ~~_/) ~~_/) ~~
*********************************************************************************
________________________________

**Comment by yarikoptic** [comment]
I think we might eventually want to work out some basic GUI app using smth like PyQt5 to optionally expose all the functionality we have exposed in CLI via GUI, at least for `upload` to start with.   Filed a dedicated https://github.com/dandi/dandi-cli/issues/835

**Comment by susiefeng** [comment]
Any update on this? user from 2024 is really looking forward to this feature

**Reply by CodyCBakerPhD** [reply]
Hi @susiefeng!

The recent beta release of the [NWB GUIDE](https://nwb-guide.readthedocs.io/en/latest/) desktop application supports a graphical interface for easy uploading to DANDI (among many other operations)
